core/opendaq/device/include/opendaq/device_impl.h

```diff
@@ -1096,20 +1096,28 @@ void GenericDevice<TInterface, Interfaces...>::serializeCustomObjectValues(const
 
     DeviceInfoPtr deviceInfo;
     checkErrorInfo(this->getInfo(&deviceInfo));
-    if (!forUpdate)
+    if (deviceInfo.assigned())
     {
-        if (deviceInfo.assigned())
-        {
-            serializer.key("deviceInfo");
-            deviceInfo.serialize(serializer);
-        }
+        serializer.key("deviceInfo");
+        deviceInfo.serialize(serializer);
+    }
```

Comment:
I don't think we need to serialize the DeviceInfo? We probably only need parts of it.